app/code/Magento/Inventory/Indexer/StockItem/GetPartialReindexData.php

```diff
@@ -70,6 +78,7 @@ public function execute(array $sourceItemIds): array
             )->where('source_item.source_item_id IN (?)', $sourceItemIds)
             ->group(['stock_source_link.' . StockSourceLink::STOCK_ID]);
 
+        $connection->query('SET group_concat_max_len = ' . $this->groupConcatMaxLen);
```

Comment:
It is better to encapsulate this logic into separate class

Reply:
Hi @naydav, how understand it is necessary to count the number of records in the table inventory_source_item, then whether to make a request for a count each time the function execute is called? In which class is it worth doing?